File: services/auth/src/main/java/com/example/iter/auth/dto/response/AdminUserStatusResponse.java
package com.example.iter.auth.dto.response;

import com.example.iter.auth.domain.entity.User;
import com.example.iter.common.security.UserStatus;

import java.time.LocalDateTime;

public record AdminUserStatusResponse(
        Long userId,
        UserStatus status,
        LocalDateTime updatedAt
) {
    public static AdminUserStatusResponse from(User user) {
        return new AdminUserStatusResponse(
                user.getId(),
                user.getStatus(),
                user.getUpdatedAt()
        );
    }
}

File: services/auth/src/main/kotlin/com/example/iter/auth/dto/response/AdminUserStatusResponse.kt
package com.example.iter.auth.dto.response

import com.example.iter.auth.domain.entity.User
import com.example.iter.common.security.UserStatus
import java.time.LocalDateTime
import kotlin.jvm.JvmRecord

@JvmRecord
data class AdminUserStatusResponse(val userId: Long?, val status: UserStatus?, val updatedAt: LocalDateTime?) {
    companion object {
        @JvmStatic
        fun from(user: User): AdminUserStatusResponse = AdminUserStatusResponse(
            user.id,
            user.status,
            user.updatedAt
        )
    }
}
